src/main/java/frc/robot/RobotContainer.java

```diff
@@ -1,145 +1,176 @@
-  public final VisionSubsystem m_VisionSubsystem = new VisionSubsystem();
-
-// Joysticks
-  private final CommandGenericHID m_driverHID = new CommandGenericHID(0);
-  
-
-// Commands
-  
-  private final TankDriveSubsystem.driveMotorCommand m_driveCommand = m_tankDriveSubsystem. new driveMotorCommand(
-    () -> m_driverHID.getRawAxis(1), 
-    () -> m_driverHID.getRawAxis(5),
-    m_tankDriveSubsystem
-    );
-
-  SendableChooser<Command> m_autoChooser = new SendableChooser<>();
-  SendableChooser<String> m_driverChooser = new SendableChooser<>();
-  
-  
-  
-
-  // m_controllerType 0 -> Unrecognized
-  // m_controllerType 1 -> Xbox Controller
-  // m_controllerType 2 -> Playstation Controller
-  // m_controllerType 3 -> Flight Joystick
-  int m_controllerType;
-
-  // Triggers
-
-  /**
-  * The container for the robot.  Contains subsystems, OI devices, and commands.
-  */
-  private RobotContainer() {
-    //m_driverChooser.setDefaultOption("DEFAULT", null);
-    //m_driverChooser.addOption("Caleb", null);
-
-
-    configureButtonBindings();
-
-    int pairButton;
-    String hidType = m_driverHID.getHID().getName();
-    if (hidType.equals("")) { // Xbox Controller | Name Unknown
-      m_controllerType = 1;
-      pairButton = 7;
-    } else if (hidType.equals("Wireless Controller")) { // PS5 | Is still called "Wireless Controller" if plugged in with a wire.
-      m_controllerType = 2;
-      pairButton = 7;
-    } else {
-      m_controllerType = 0;
-      pairButton = 7;
+    private static RobotContainer m_robotContainer = new RobotContainer();
+    // Subsystems
+    public final TankDriveSubsystem m_tankDriveSubsystem = new TankDriveSubsystem();
+    public final IntakeSubsystem m_intake = new IntakeSubsystem();
+    private static MiscDashboardSubsystem m_miscDashboardSubsystem = new MiscDashboardSubsystem();
+
+    // other stuff
+    private final CommandGenericHID m_driverHID = new CommandGenericHID(0);
+    private final CommandGenericHID m_operatorHID = new CommandGenericHID(1);
+
+    // Commands
+    private final TankDriveSubsystem.driveMotorCommand m_driveCommand = m_tankDriveSubsystem.new driveMotorCommand(
+            () -> m_driverHID.getRawAxis(1),
+            () -> m_driverHID.getRawAxis(5),
+            m_tankDriveSubsystem);
+
+    private final TankDriveSubsystem.driveMotorCommand m_operatorDrive = m_tankDriveSubsystem.new driveMotorCommand(
+            () -> m_operatorHID.getRawAxis(1),
+            () -> m_operatorHID.getRawAxis(5),
+            m_tankDriveSubsystem);
+
+    // Drivetrain is reversed when button A is pressed on the controller:
+    SwitchDrivetrainInvert m_SwitchDrivetrainInvertCommand = m_tankDriveSubsystem.new SwitchDrivetrainInvert(
+            m_tankDriveSubsystem);
+    Trigger invertDrivetrainTrigger = m_driverHID.button(1)
+            .onTrue(m_tankDriveSubsystem.new SwitchDrivetrainInvert(m_tankDriveSubsystem));
```

Comment:
I believe m_SwitchDrivetrainInvertCommand is unused here. We just instantiate a new SwitchDrivetrainInvert on the following line.

Reply:
Also, does this do what we want?
Consider moving this down where all other commands are linked. Consider following the same design pattern there of:

```
Trigger invertDrivetrainTrigger = m_driverHID.button(1);
invertDrivetrainTrigger.onTrue(m_tankDriveSubsystem.new SwitchDrivetrainInvert(m_tankDriveSubsystem));
```